### PR TITLE
Add support for `TableField`

### PR DESCRIPTION
**Pull request by keichinger**

### CODE REVIEW

File: src/Field/Data/TableData.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		);
+	}
+
+	private function flattenRows (array $row) : array
+	{
+		return \array_map(
+			static fn (array $column) => $column["value"],
+			$row,
+		);
+	}
```
Comment (apfelbox) [inline src/Field/Data/TableData.php]:
```suggestion
		return \array_column($row, "value");
	}
```

File: src/Field/Data/Table/TableRowData.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php declare(strict_types=1);
+
+namespace Torr\Storyblok\Field\Data\Table;
+
+final class TableRowData
```
Comment (apfelbox) [inline src/Field/Data/Table/TableRowData.php]: Do we need that VO?

Comment (keichinger) [reply src/Field/Data/Table/TableRowData.php]: Potentially. A full table row (within the `tbody`) contains more meta data such as field type and the corresponding uid for the row.

If we don't want it, I wouldn't mind removing the row VO and just work with raw `TableColumnData` arrays.

File: src/Field/Data/TableData.php (hunk 1 above)
Comment (keichinger) [reply src/Field/Data/TableData.php]: I'm always a bit hesitant with the usage of `array_column` as it sometimes seems confusing and is one of those methods that feel overly complex.

Comment (apfelbox) [reply src/Field/Data/TableData.php]: But for this use case it is pretty easy and straight forward. 😄

File: src/Field/Data/Table/TableRowData.php (hunk 2 above)
Comment (apfelbox) [reply src/Field/Data/Table/TableRowData.php]: I would remove it for now. We can always expand later.

File: src/Field/Definition/TableField.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	{
+		$context->ensureDataIsValid(
+			$contentPath,
+			$this,
+			$data,
+			[
+				!$this->allowMissingData && $this->required ? new NotNull() : null,
+				new Type("array"),
+				// We can't validate the length here, as it is not guaranteed if you add
+				// the max-length after content was added.
```
Comment (apfelbox) [inline src/Field/Definition/TableField.php]: We validate the structure with `thead` and `tbody` etc here.

**Review by apfelbox — CHANGES_REQUESTED** [review]

File: src/Field/Definition/TableField.php (hunk 3 above)
Comment (keichinger) [reply src/Field/Definition/TableField.php]: I've added some validation 👍

**Pull request merged**